clientconn.go

```diff
@@ -152,6 +152,25 @@ func NewClient(target string, opts ...DialOption) (conn *ClientConn, err error)
 	for _, opt := range opts {
 		opt.apply(&cc.dopts)
 	}
+
+	// Register ClientConn with channelz.
+	cc.channelzRegistration(target)
+	// TODO: Ideally it should be impossible to error from this function after
```

Comment:
This diff makes the problem this TODO describes even worse.  And there are places where we error from this now that don't call `channelz.RemoveEntry`, leaking resources.

Please make `parseTargetBlah` stop doing channelz logs (maybe move some useful channelz messages into this function after channelz registration), and move channelz registration back down to where it was.

If you're feeling extra helpful, you could also extend that to the other initialization functions that might do channelz logs, and make sure we don't create the channelz representation of this channel until immediately before the function returns.

Reply:
Switched to solution discussed offline; make parseTarget log at info, and log authority in NewClient body. Only create channelz after we know any errors can occur.